### DISCUSSION TITLE
How to Get dSYM Files for Crash Reporting

### DISCUSSION THREAD

**Discussion by anthonydito** [bazelbuild/rules_apple#1767] [Q&A] [answered]
I have been trying for a few days to get dSYM files for crash reporting of my iOS app built with Bazel. For reference, the build command that I am using is.

```
bazel build //ios:ios_prod --apple_generate_dsym --ios_multi_cpus=arm64 --define=apple.add_debugger_entitlement=no --output_groups=+dsyms
```

Now, in the `bazel-out` folder. I do see a `fourier.app.dSYM` file. However, I don't see a dSYM files that I need according to Firebase Crashlytics (which are associated with a UUID).

My question is: With Bazel, how can I get the dSYM files that I need for Crashlytics? Am I missing something that is outputted by Bazel but I need to extract?

**Comment by thii** [comment, marked as answer]
The `fourier.app.dSYM` bundle you see should be all needed for crash reporting. What else are you looking for specifically?

**Reply by anthonydito** [reply]
See the below image from Firebase about my missing dSYM files.

<img width="942" alt="Screenshot 2022-11-28 at 8 34 23 AM" src="https://user-images.githubusercontent.com/8814826/204331335-788604d4-1860-45d1-9070-fc61a6e5c902.png">

In other apps, when I do a standard build process using XCode, these dSYM files exist in the filesystem with the UUID. That was what I was specifically looking for here. And they don't appear to be in the `fourier.app.dSYM`

**Reply by keith** [reply]
you can inspect the dsym / binary UUIDs by running `dwarfdump -u path/to/binary` to make sure they match. if those UUIDs are the ones you see here, they could be referring to something else like a third party dynamic framework dependency that you need to upload dsyms from your vendor for

**Reply by anthonydito** [reply]
Thank you! This was quite dumb of me. Simply renaming the `fourier.app.dSYM` to the name needed by Crashlytics, zipping, and uploading, worked for me.
